### PR TITLE
Pause state setter for plugin API

### PR DESCRIPTION
**Pull request by spacek531**
It is impossible to set pause state on loading without losing 1 frame when hooked to `map.changed` and it is impossible to set pause state before park save when hooked to `map.save`. This PR fixes that by allowing plugins to set pause state directly.

Setter is disabled in networked mode because it is not network-safe to call PauseToggle without extra stuff that the game action does perfectly well.

### CODE REVIEW

**Comment by spacek531** [comment]
You can try this feature out with this plugin: https://github.com/spacek531/ParkPauser/blob/master/src/ParkPauser.js

**Comment by Sadret** [comment]
I'm not sure if we should introduce even more unsafe game state manipulations like this one, but it does not look too problematic.
I think your use case is niche but valid.

File: src/openrct2/scripting/bindings/game/ScContext.hpp [hunk 1]
```diff
@@ -131,6 +131,21 @@ namespace OpenRCT2::Scripting
...
+        {
+#    ifndef DISABLE_NETWORK
+            if (NetworkGetMode() != NETWORK_MODE_NONE)
+            {
+                auto ctx = GetContext()->GetScriptEngine().GetContext();
+                duk_error(
+                    ctx, DUK_ERR_ERROR, "Setting paused state is not network safe. Use the pausetoggle game action instead.");
+                return;
+            }
+#    endif
```
Comment (Basssiiie) [inline src/openrct2/scripting/bindings/game/ScContext.hpp:144]: Is there a reason to not use `ThrowIfGameStateNotMutable()` that all other setters use to check for unsafe state modifications?

Comment (spacek531) [reply src/openrct2/scripting/bindings/game/ScContext.hpp:144]: I didn't know about that method, I will check it out

Comment (spacek531) [reply src/openrct2/scripting/bindings/game/ScContext.hpp]: Wouldn't I know it but the method does exactly what I programmed. It is implemented.

**Review by Basssiiie — APPROVED** [review]
Sorry for the delay but I have tested it and it all looks good to me! 😄

**Review by janisozaur — APPROVED** [review]

**Comment by spacek531** [comment]
Rebased

**Comment by spacek531** [comment]
Are there any current blockers to merging this?

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**